enhancements/network/user-space-pod-iface-and-api-library.md

```diff
@@ -0,0 +1,262 @@
+appear that there were any major objections to the PR, just lack of activity.
+
+If this feature can make it into upstream Kubernetes, then the pod will have
+hugepage information needed by user space applications.
+
+#### Additional Interface Data
+
+Work is being done in the Network Plumbing Working Group (NPWG) to standardize
+how data for secondary interfaces is passed to a pod. NPWG is where the standard
+is defined that Multus is implemented too. See:
+[Kubernetes Network Custom Resource Definition De-facto Standard](https://docs.google.com/document/d/1Ny03h6IDVy_e_vmElOqR7UdTPAG_RNydhVE1Kx54kFQ/edit?usp=sharing).
+The '*k8s.v1.cni.cncf.io/networks-status*' field is being expanded to include
+additional data for each interface, specific to the interface type data. See
+Proposal:
+
+[NPWG - Device Information](https://docs.google.com/document/d/1rBm-L1ymXIjoKNA6w2lixwBAjt9-tnbs-ee7AcLRbVE/edit?usp=sharing)
+
+The proposal has been discussed in the NPWG for several months and there is
+general agreement on the feature and now work is being done on finalizing the
+smaller details, The remaining work includes:
+- Updating the specification, defining the new fields in
+'*k8s.v1.cni.cncf.io/networks-status*'
+- Completing a best practices document which describes how the data is passed
+between a Device Plugin or Admission Controller to a CNI, and what Multus needs
+to do with the collected data.
+
+Once that is completed, the following projects will need to be updated to add
+support (proof of concept already exists for most of these):
+- SR-IOV Device Plugin
+- SR-IOV CNI
+- SR-IOV Network-Resource Injector
+- Multus
+- app-netutil
+
+#### Consumption of Data in the Pod
+
+Data passed into a pod can come from several places. Below are some examples:
+- For kernel interfaces, data can be applied to the interface via the kernel
+itself.
+- For SR-IOV VFs, the PCI Address list is passed into the pod via an
+environmental variable.
+- Using Kubernetes Downward API, pod annotation data is exposed within the pod.
+Data is written to the pod annotation and consumed by the pod.
+
+To ease the burden of the CNF developer, an API library is being developed that
+collects data from all the different sources and exposes the data through a set
+of APIs. The API is written in Golang and can be imported into other Golang
+programs. Since DPDK is the primary user space application being targeted, and
+DPDK is written in C, there is also a C API binding for the Golang library.
+
+The library is located in github at the following URL:
+- https://github.com/openshift/app-netutil/
+
+The library will not be released in OCP, but will be available for CNF
+developers to include in their projects. If the CNF developers choose not to
+include the library, the library will still serve as an example of where to pull
+the data from and how to consume the data.
+
+The following APIs will be exposed:
+- GetCPUInfo()
```

Comment:
general style note: use backticks around things that are "code-like" (eg, function names here, references to API fields, annotation names, etc) a lot of the places where you're currently doing `"*foo*"` would be better as `` `foo` ``

Reply:
Done